### ISSUE TITLE
IOException "Write dead end" using pipes.shell/exec

### ISSUE THREAD

**Issue by dgoeke** [bilus/pipes#1]
Hi Bilus,

Thanks for making this!  I'm having trouble using it, but I'm not sure if it's my error or a problem with the library.  Here's my code:

``` clojure
(ns demo.pipes
  (:require [pipes.core :refer [pipe-> ->pipe-> ->job->]]
            [pipes.shell :refer [exec]]
            [clojure.java.io :as io]
            [clojure.string :as str])
  (:import [java.io ByteArrayInputStream
            ByteArrayOutputStream]))

(def identity-job
  (->job-> [in out] (io/copy in out)))

(defn test-works-1 []
  (with-open [in  (ByteArrayInputStream. (.getBytes "yes yes yes\nno no no"))
              out (ByteArrayOutputStream.)]
    @(->pipe-> in
               [identity-job
                (exec ["grep" "no"])]
               out)
    (str out)))

(defn test-works-2 []
  (with-open [in  (ByteArrayInputStream. (.getBytes "yes yes yes\nno no no"))
              out (ByteArrayOutputStream.)]
    @(->pipe-> in
               [(exec ["grep" "no"])]
               out)
    (str out)))

(defn test-fails []
  (with-open [in  (ByteArrayInputStream. (.getBytes "yes yes yes\nno no no"))
              out (ByteArrayOutputStream.)]
    @(->pipe-> in
               [(exec ["grep" "no"])
                identity-job]
               out)
    (str out)))
```

So we define three tests, all using `->pipe->`:
- `test-works-1` sends the data through `[identity-job (exec ...)]` and succeeds
- `test-works-2` removes the identity job and only uses `[(exec ...)]` and succeeds
- `test-fails` just reverses the order, using `[(exec ...) identity-job]` and fails with a IOException "Write dead end"

I was expecting all 3 of these to produce the same result.

**Comment by bilus** [comment]
I'm sorry for the delay. I'll try to help you out over the weekend.

**Comment by bilus** [comment]
Ouch, ok, fixed that after 1 hour of debugging. I extracted the code from a production app and while cleaning up the code etc. I toggled the `close-out?` flag in `exec` code. After the fix it works.

I'll push out 0.1.2 in a sec. 

Thanks a lot for letting me know about the issue and I'm very sorry it took me this long to get back with a fix but things are crazy at work (in a positive sense, just super busy) and this is literally the first day I could look at what's going on my open source projects.
